locust/runners.py

```diff
@@ -490,6 +491,8 @@ def client_listener(self):
                 if msg.node_id in self.clients:
                     del self.clients[msg.node_id]
                     logger.info("Client %r quit. Currently %i clients connected." % (msg.node_id, len(self.clients.ready)))
+                    if self.worker_count == 0:
+                        self.stop()
```

Comment:
We could log a message here. `Last worker quit. Stopping test` or something.

Reply:
Isnt that implied by the last ”client x quit. Currently 0 clients connected” though?

Reply:
Side note: I dont like the word ”clients” for load gen workers, as there is a chance of confusion with test clients/users. That is a different issue..

Reply:
I think this log message would be helpful.

Also, I was thinking I should add a comment here to clarify that calling stop here isn't responsible for stopping the client_listener greenlet, and mention where it is being stopped/killed. Buuut I'm not totally sure what IS stopping it when the last worker quits?

Reply:
> Isnt that implied by the last ”client x quit. Currently 0 clients connected” though

Maybe, but I don't think it hurts to be extra clear about it. One could perhaps expect that the master would re-balance the simulated users and the test would continue to run if a new worker connects (just as would happen if all workers except one quits, and then a new worker connects).

> Side note: I dont like the word ”clients” for load gen workers

Definitely agree! It should say workers.

> Also, I was thinking I should add a comment here to clarify that calling stop here isn't responsible for stopping the client_listener greenlet

The client_listener is not supposed to end when workers disconnect, so I don't think that's necessary.

Reply:
> The client_listener is not supposed to end when workers disconnect, so I don't think that's necessary.

Oh okay, I see what's happening now. It's just hanging when it tries to recv_from_client() since there are no more workers to receive from.